core/src/main/scala/Main.scala

```diff
@@ -1,21 +1,19 @@
 import zio.{Console, Scope, ZIO, ZIOAppDefault, ZLayer}
 
 object Main extends ZIOAppDefault:
+  val program: ZIO[CarApi, Nothing, Unit] =
```

Comment:
@kitlangton One problem I see here is that we are starting to use environment for yet another purpose, which I'm not sure I can capture as a clear rule :) Previously env was used for contextual values - not for passing dependencies. I know it's nit-picking, but I wanted the example to demonstrate some clear rules :)

So I'd expect sth like `def program(api: CarApi): ZIO[Any, Nothing, Unit]`. Or maybe we should extend the use-cases of the environment to cover "entry point to the application"? After all, we do need to kick things off somehow.